docs/standard/analyzers/api-analyzer.md

```diff
@@ -0,0 +1,69 @@
+---
+title: .NET API Analyzer
+description: Learn how the .NET API Analyzer can help detecting deprecated APIs and platform compatibility issues.
+author: oliag
+ms.author: mairaw
+ms.date: 12/11/2017
+ms.topic: article
+ms.prod: .net
+ms.technology: dotnet-standard
+---
+
+# .NET API Analyzer
+
+The .NET API Analyzer is a Roslyn analyzer that discovers potential compatibility risks for APIs with different platforms and detects calls to deprecated APIs. It can be useful for any developers of the .NET family at any stage of development.
+
+API Analyzer comes as a NuGet package [Microsoft.DotNet.Analyzers.Compatibility](https://www.nuget.org/packages/Microsoft.DotNet.Analyzers.Compatibility/) and after you reference it in a project, it automatically monitors the code and indicates problematic API usage. You can also get suggestions on possible fixes by clicking on the light bulb. The drop-down menu includes an option to suppress the warnings.
+
+## Discovering deprecated APIs
+
+### *What is deprecated API*
+*The .NET Framework is a large product that is getting constantly upgraded to better serve customer needs. It is natural to deprecate some APIs and replace them with new ones. Previously old APIs were marked with `[Obsolete]` attribute. There was only one diagnostic ID (CS0612) for all obsolete APIs that lead to one description for all cases and ability to suppress warnings for either all cases or none of them. To provide information and a way to suppress a warning per each API or a class of APIs, `Deprecation` term was introduced.*
```

Comment:
There are a number of problems with this paragraph:
- CS0612 is a compiler-specific error message. (The VB code is BC40000.) Does API Analyzer only support C# code right now? If so, that's not mentioned. If not, this needs to be made more language-independent. For example: "Because compilers used a single error ID (for C#, CS0612; for Visual Basic, BC40000) for all obsolete APIs, developers had the limited choice of suppressing warnings for all cases or for none." 
- "old APIs" is ambiguous; are old APIs deprecated, or have they just been around a long time? It's best to describe them as "deprecated APIs", which maintains consistency with the second sentence.
- "Previously, old APIS were marked..." implies that the Obsolete attribute is no longer used to indicate new deprecated APIs. Is that true?
- The last sentence is confusing. I think that the point you're trying to make is that API Analyzer uses API-specific error codes that begin with DE (for Deprecation Error?), which allows control over the display of individual warnings. 
In addition, some other smaller comments:
- is getting constantly upgraded --> is constantly upgraded
- comma after Previously

Reply:
API Analyzer support only C# code right now but we are looking at ways to extend it to VB as well.

---
In reply to: [160496715](https://github.com/dotnet/docs/pull/3908#discussion_r160496715) [](ancestors = 160496715)